proposals/NNNN-swiftpm-html-coverage-report.md

```diff
@@ -0,0 +1,185 @@
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+*During the review process, add the following fields as needed:*
+
+* Implementation: [swiftlang/swift-package-manager#9076][PR]
+* Decision Notes: [Pitch](https://forums.swift.org/t/pitch-adding-html-coverage-support/82358)
+
+## Introduction
+
+Currently, `swift test` supports generating a JSON coverage report, which is
+great for ingesting into various systems. The JSON, however, is not very "human
+consumable friendly" while iterating at-desk.
+
+This proposes adding an additional command line argument to `swift test` that
+would allow the caller to select the generation of an HTML coverage report.
+
+
+## Motivation
+
+JSON coverage report is great for ingesting into exgernal tools that post-process
+the coverage data.  Allowing users to generate an HTML coverage report allows
+the HTML coverage report to be uploaded to CI system for visual inspection, and
+even allows users to generate HTML coverage report at-desk, allow faser feedback
+to determine if the current changes are sufficiently covered to their liking.
+
+## Proposed solution
+
+if users currently want an HTML report, custom scripts are written to get access
+to the coverage data, and then invoking the `llvm-cov` binary directly
+
+e.g.:
+
+```
+❯ swift test --enable-code-coverage
+
+❯ swift test --show-codecov-path
+
+❯ llvm-cov show \
+  --project-title="HelloWorld" \
+  --format="html" \
+  --output-dir=".coverage" \
+  --instr-profile=".build/arm64-apple-macosx/debug/codecov/default.profdata" \
+  ".build/.../HelloWorldPackageTests.xctest/Contents/MacOS/HelloWorldPackageTests" \
+  "Sources"
+```
+
+Since SwiftPM currently orchestrates the JSON coverage data, the solution adds a
+new command line argument, e.g.: `--coverage-format` to `swift test` which can 
+be specified multiple times, to generate multiple coverage report type from a
+single test execution.
+
+While processing the coverage data, SwiftPM will loop on all the unique coverage
+format options to generate the specified reports.
+
+Unless otherwise specified, this proposal applies only to the HTML Coverage
+report.  The generation of the JSON Coverage report is unchanged and is out of
+scope.
+
+## Detailed design
+
+Existing tools in LLVM have been around for several years (and maybe even decades), 
+and provides robust tools for code coverage analysis. The LLVM tools are
+well-documented, and have been used in production for many years.  SwiftPM will
+make use of this and contruct the proper command line arguments to the
+`llvm-cov show` utility, which will generate the HTML report.
+
+### Format Selection
+
+The `swift test` command line will have an option named `--coverage-format`,
+which accepts either `json` or `html`.  This option can be specified multipe
+times on the command line, and a report will be generated for each format
+specified.
+
+The command line option will be defined as
+
+```swift
+    @Option(
```

Comment:
You don't need to include the source code you're adding to SwiftPM here. Just describe the command-line argument in terms of what the _user_ will deal with (e.g. show the console output for it when you write `swift test --help`).